### PR TITLE
Optimizing Utils.findProjectByParameterUUID

### PR DESCRIPTION
**Pull request by jglick**
Observed a Jenkins instance freezing up with this in the support bundle:

```
49761msec elapsed in Handling GET /jenkins/job/…/build from "…" : … ParametersDefinitionProperty/index.jelly CascadeChoiceParameter/index.jelly CascadeChoiceParameter/choiceParameterCommon.jelly CascadeChoiceParameter/selectContent.jelly
	java.lang.String$CaseInsensitiveComparator.compare(String.java:1186)
	java.lang.String.compareToIgnoreCase(String.java:1239)
	hudson.util.CaseInsensitiveComparator.compare(CaseInsensitiveComparator.java:40)
	hudson.util.CaseInsensitiveComparator.compare(CaseInsensitiveComparator.java:34)
	jenkins.model.IdStrategy$CaseInsensitive.compare(IdStrategy.java:176)
	jenkins.model.IdStrategy.equals(IdStrategy.java:90)
	nectar.plugins.rbac.groups.Group.isMatch(Group.java:493)
	nectar.plugins.rbac.groups.Group.isMatch(Group.java:445)
	nectar.plugins.rbac.groups.Group.isMatch(Group.java:382)
	nectar.plugins.rbac.groups.GroupContainerACL.hasPermission(GroupContainerACL.java:196)
	nectar.plugins.rbac.groups.GroupContainerACL._hasPermission(GroupContainerACL.java:123)
	nectar.plugins.rbac.groups.GroupContainerACL.hasPermission(GroupContainerACL.java:76)
	hudson.security.ACL.hasPermission(ACL.java:78)
	hudson.model.AbstractItem.hasPermission(AbstractItem.java:495)
	com.cloudbees.hudson.plugins.folder.AbstractFolder.getItems(AbstractFolder.java:1146)
	hudson.model.Items.getAllItems(Items.java:407)
	hudson.model.Items.getAllItems(Items.java:417)
	hudson.model.Items.getAllItems(Items.java:417)
	hudson.model.Items.getAllItems(Items.java:417)
	hudson.model.Items.getAllItems(Items.java:417)
	hudson.model.Items.getAllItems(Items.java:403)
	jenkins.model.Jenkins.getAllItems(Jenkins.java:1758)
	org.biouno.unochoice.util.Utils.findProjectByParameterUUID(Utils.java:166)
	org.biouno.unochoice.AbstractScriptableParameter.getHelperParameters(AbstractScriptableParameter.java:171)
	org.biouno.unochoice.AbstractScriptableParameter.eval(AbstractScriptableParameter.java:236)
	org.biouno.unochoice.AbstractScriptableParameter.getChoices(AbstractScriptableParameter.java:200)
	org.biouno.unochoice.AbstractScriptableParameter.getChoices(AbstractScriptableParameter.java:188)
```

The method is searching for a project according to some criteria. This stack is pretty silly for two reasons:

* You are performing an access control check, which can be expensive, on every project (or whatever) in the system, before even looking for the match.
* You are sorting the results by name, only to iterate them and look for just one.

2.37+ APIs allow for a more efficient route.

@reviewbybees

### CODE REVIEW

**Comment by kinow** [comment]
+1 ! In the beginning there was no access control, just returning the project. But then came requests to find the whatever (some project types provided by plug-ins, or in a different hierarchy), and only then the access check. Silly mistake nevertheless. Thanks for fixing that!

Happy to merge, but guess the proper process is to wait someone else from CloudBees to review it?

**Comment by jglick** [comment]
> only then the access check

Note that the access check in `master` is being done implicitly, by `Jenkins.getAllItems(Class)`. I am retaining that check (since I am unsure of exactly how the utility methods are being used), but moving it until _after_ we find a match.

> Happy to merge, but guess the proper process is to wait someone else from CloudBees to review it?

Not at all. You are the maintainer, merge whenever you like:

> please see [this explanation](https://github.com/reviewbybees/about#about-reviewbybees)

That said, I filed #18 to let you see in the PR whether tests pass. I did not bother running them myself, because, hey, Jenkins?

File: pom.xml [hunk 1]
```diff
@@ -109,17 +110,17 @@
             <version>2.0.2-beta</version>
             <scope>test</scope>
         </dependency>
-        <dependency>
+        <dependency> <!-- TODO why is this here? -->
```
Comment (kinow) [inline pom.xml]: Either I added this dependency for some test that was failing, or I copied it from another plug-in while updating the pom.xml (very likely). Happy to remove it if not necessary (i.e. if build passes without it, or dependency tree shows it's pointless to include it)

Comment (jglick) [reply pom.xml]: Not sure offhand. Leaving it there for purposes of this PR.

**Comment by jglick** [comment]
There, looks like I got the build to pass.

**Comment by kinow** [comment]
Thanks ton for the Jenkinsfile and for fixing this one too Jesse.

**Pull request merged**